### PR TITLE
[DBP-360] Updating scripts to reflect PG-16 paths and configurations …

### PR DESCRIPTION
**Pull request by agarwal-kritika**
…on macOS

Bump PG version PG-16beta1

### CODE REVIEW

File: server/build-osx.sh [hunk 1]
```diff
@@ -199,7 +199,8 @@ _build_server_osx() {
     # Building 'System_stats' extention and bundled the required files with posgresql
     ssh $PG_SSH_OSX "cd $PG_PATH_OSX/server/source/system_stats.osx; PATH="$PG_STAGING/bin:$PATH" make USE_PGXS=1; PATH="$PG_STAGING/bin:$PATH" make install USE_PGXS=1" || _die "Failed to build System Status"
 
-    ssh $PG_SSH_OSX "cd $PG_PATH_OSX/server/source/system_stats.osx/; cp system_stats--*.sql system_stats.control $PG_STAGING/share/postgresql/extension; cp system_stats.so $PG_STAGING/lib/postgresql " || _die "Failed to bundle system_stats extension files"
+    #ssh $PG_SSH_OSX "cd $PG_PATH_OSX/server/source/system_stats.osx/; cp system_stats--*.sql system_stats.control $PG_STAGING/share/postgresql/extension; cp system_stats.so $PG_STAGING/lib/postgresql " || _die "Failed to bundle system_stats extension files"
```
Comment (sandeep-edb) [inline server/build-osx.sh]: please remove commented lines

File: server/build-osx.sh [hunk 2]
```diff
@@ -199,7 +199,8 @@ _build_server_osx() {
...
+     ssh $PG_SSH_OSX "cd $PG_PATH_OSX/server/source/system_stats.osx/; cp system_stats--*.sql system_stats.control $PG_STAGING/share/postgresql/extension " || _die "Failed to bundle system_stats extension files"
```
Comment (sandeep-edb) [inline server/build-osx.sh]: either .so or .dylib should be copied, which is missing here

File: server/build-osx.sh [hunk 3]
```diff
@@ -243,6 +244,14 @@ _build_server_osx() {
...
     ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libiconv*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libiconv"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libkrb5*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libkrb5"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libgss*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libgssapi"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libk5*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libk5"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libverto*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libverto"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libcom*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libcom"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libkdb5*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libkdb5"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libkadm5*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libkadm5"
+    ssh $PG_SSH_OSX "cp -pR /opt/local/Current_v15/lib/libkrad*.dylib $PG_STAGING/lib/" || _die "Failed to copy the latest libkrad"
```
Comment (sandeep-edb) [inline server/build-osx.sh]: hope all these dylibs are required and we are not necessarily bundling the ones which are not needed?
please also update 3rd party license to include krb5

File: server/pgserver.xml.in [hunk 4]
```diff
@@ -325,7 +325,7 @@
         <!-- MAC: Stop the server -->
         <runProgram>
             <program>su</program>
-            <programArguments>- ${serviceaccount} -c "\"${installdir}/bin/pg_ctl\" stop -m fast -w -D \"${datadir}\""</programArguments>
+            <programArguments>- ${serviceaccount} -c "\"${installdir}/bin/postgres\" stop -m fast -w -D \"${datadir}\""</programArguments>
```
Comment (sandeep-edb) [inline server/pgserver.xml.in]: any specific reason to change this?

File: server/pgserver.xml.in [hunk 5]
```diff
@@ -339,7 +339,7 @@
         </runProgram>
         <runProgram>
             <program>launchctl</program>
-            <programArguments>unload /Library/LaunchDaemons/com.edb.launchd.${servicename}.plist</programArguments>
+            <programArguments>unload /Library/LaunchDaemons/${servicename}.plist</programArguments>
```
Comment (sandeep-edb) [inline server/pgserver.xml.in]: any specific reason to change this? Usually all the .plist file begins with **com.**

**Comment by sandeep-edb** [comment]
@agarwal-kritika also, please update https://github.com/EnterpriseDB/edb-installers/blob/REL-16/server/resources/installation-notes.html

File: server/pgserver.xml.in (hunk 5 above)
Comment (agarwal-kritika) [reply server/pgserver.xml.in]: when i use this name with name starting with "com", it is giving error- No such fie or directory. When i gave this it ran fine

File: server/build-osx.sh (hunk 2 above)
Comment (agarwal-kritika) [reply server/build-osx.sh]: Adding that

File: server/pgserver.xml.in (hunk 4 above)
Comment (agarwal-kritika) [reply server/pgserver.xml.in]: changed back to pg_ctl

File: server/pgserver.xml.in (hunk 5 above)
Comment (sandeep-edb) [reply server/pgserver.xml.in]: but we have been using the previous name for a long time and it always worked

**Pull request merged**